client/lib/i18n-utils/switch-locale.js

```diff
@@ -63,6 +62,12 @@ export default function switchLocale( localeSlug ) {
 				return;
 			}
 
+			// Handle race condition when we're requested to switch to a different
+			// locale while we're in the middle of request, we should abondon result
+			if ( localeSlug !== switchLocaleRequestStack[ switchLocaleRequestStack.length - 1 ] ) {
```

Comment:
We probably don't need a "stack" here, just the `currentLocale` as a variable outside `switchLocale`.
```js
currentLocale = localeSlug;
request.get( languageFileUrl( localeSlug ) ).end( function( error, response ) {
  ....
  if ( localeSlug !== currentLocale ) {
    return;
  }
}
```